app/controllers/api/base_controller/normalizer.rb

```diff
@@ -34,6 +34,8 @@ def normalize_attr(attr, value)
           normalize_array(value)
         elsif value.respond_to?(:attributes) || value.respond_to?(:keys)
           normalize_hash(attr, value)
+        elsif @req.subject == "automate_workspaces" && value =~ /^password::/
+          "password::********"
```

Comment:
This isn't the way we handle other passwords - we just omit them. Is there a special reason that they need to be handled differently?

Reply:
@imtayadeway In other cases its our password say for the provider, storage. These passwords are created by customers in the Automate Model for their own use cases (connect to IPAM server, LDAP server etc) the decide the attribute name. Sending the password this way allows the client side to check if 
* the attribute exists
* the attribute has the correct data type

Once these pre conditions have been met it can make a REST API call to decrypt the password.

Reply:
@mkanoor can you add a test for the case in which the attribute does not exist? How is the data type revealed through the response? Can you also ensure there is test coverage for the happy/sad path there also?

Reply:
@imtayadeway Thats already implemented here
https://github.com/ManageIQ/manageiq-api/blob/master/spec/requests/automate_workspaces_spec.rb#L103

If you send any bogus object/attribute pair we send back an empty string.

This PR is the about the "GET" request for Automate Workspace and filters/masks what is being sent out, its not about encrypting/decrypting passwords on demand which was done in PR.
https://github.com/ManageIQ/manageiq-api/pull/124

Reply:
I'm thinking that this may be a better candidate for inheritance - as it is, every single request is going to check if we're dealing with automate workspaces multiple times:

```ruby
# app/controllers/api/automate_workspaces_controller.rb

private

def normalize_attr(attr, value)
  return "password:***" if value /\Apassword/
  super
end
```